spec/models/reviewable_history_spec.rb

```diff
@@ -54,9 +56,62 @@
     history = reviewable.history
     expect(history.size).to eq(2)
 
-    expect(history[1].reviewable_history_type).to eq(ReviewableHistory.types[:edited])
+    expect(history[1]).to be_edited
     expect(history[1].created_by).to eq(moderator)
     expect(history[1].edited).to eq("category_id" => [old_category.id, nil])
   end
 
+  describe "Callbacks" do
+    let!(:reviewable) { Fabricate(:reviewable_queued_post) }
+    let(:user) { reviewable.created_by }
+    let(:user_stats) { user.user_stat }
+
+    before do
+      user_stats.update!(pending_posts_count: 0) # because callbacks
```

Comment:
Callbacks make it a bit hard to test in isolation things as they create a lot of side effects. Maybe this is an area that could be improved by using a pub/sub approach for example (I’m thinking about Wisper).